web/src/lib/components/asset-viewer/photo-viewer.svelte

```diff
@@ -1,102 +1,82 @@
-  import { type AssetResponseDto, AssetTypeEnum, AssetMediaSize } from '@immich/sdk';
-  import { useZoomImageWheel } from '@zoom-image/svelte';
-  import { onDestroy, onMount } from 'svelte';
+  import { getAltText } from '$lib/utils/thumbnail-util';
+  import { AssetTypeEnum, type AssetResponseDto, AssetMediaSize } from '@immich/sdk';
+  import { zoomImageAction, zoomed } from '$lib/actions/zoom-image';
+  import { canCopyImagesToClipboard, copyImageToClipboard } from 'copy-image-clipboard';
+  import { onDestroy } from 'svelte';
+
   import { fade } from 'svelte/transition';
   import LoadingSpinner from '../shared-components/loading-spinner.svelte';
   import { NotificationType, notificationController } from '../shared-components/notification/notification';
-  import { getAltText } from '$lib/utils/thumbnail-util';
-  import { SlideshowLook, slideshowLookCssMapping, SlideshowState, slideshowStore } from '$lib/stores/slideshow.store';
-
-  const { slideshowState, slideshowLook } = slideshowStore;
 
   export let asset: AssetResponseDto;
-  export let preloadAssets: AssetResponseDto[] | null = null;
+  export let preloadAssets: AssetResponseDto[] | undefined = undefined;
   export let element: HTMLDivElement | undefined = undefined;
   export let haveFadeTransition = true;
 
-  let imgElement: HTMLDivElement;
-  let assetData: string;
-  let abortController: AbortController;
-  let hasZoomed = false;
-  let copyImageToClipboard: (source: string) => Promise<Blob>;
-  let canCopyImagesToClipboard: () => boolean;
-  let imageLoaded: boolean = false;
+  export let copyImage: (() => Promise<void>) | null = null;
+  export let zoomToggle: (() => void) | null = null;
 
-  const loadOriginalByDefault = $alwaysLoadOriginalFile && isWebCompatibleImage(asset);
-
-  $: if (imgElement) {
-    createZoomImageWheel(imgElement, {
-      maxZoom: 10,
-      wheelZoomRatio: 0.2,
-    });
-  }
-
-  onMount(async () => {
-    // Import hack :( see https://github.com/vadimkorr/svelte-carousel/issues/27#issuecomment-851022295
-    // TODO: Move to regular import once the package correctly supports ESM.
-    const module = await import('copy-image-clipboard');
-    copyImageToClipboard = module.copyImageToClipboard;
-    canCopyImagesToClipboard = module.canCopyImagesToClipboard;
-  });
+  const { slideshowState, slideshowLook } = slideshowStore;
 
-  $: void loadAssetData({ loadOriginal: loadOriginalByDefault, checksum: asset.checksum });
+  let assetFileUrl: string = '';
+  let imageLoaded: boolean = false;
+  let imageError: boolean = false;
+  let forceUseOriginal: boolean = false;
+
+  $: isWebCompatible = isWebCompatibleImage(asset);
+  $: useOriginalByDefault = isWebCompatible && $alwaysLoadOriginalFile;
+  $: useOriginalImage = useOriginalByDefault || forceUseOriginal;
+  // when true, will force loading of the original image
+  $: forceUseOriginal = forceUseOriginal || ($photoZoomState.currentZoom > 1 && isWebCompatible) ? true : false;
```

Comment:
This ternary seems redundant

```suggestion
  $: forceUseOriginal = forceUseOriginal || ($photoZoomState.currentZoom > 1 && isWebCompatible);
```